src/ripple/rpc/handlers/ServerInfo.cpp

```diff
@@ -22,13 +22,298 @@
 #include <ripple/app/reporting/P2pProxy.h>
 #include <ripple/json/json_value.h>
 #include <ripple/net/RPCErr.h>
+#include <ripple/protocol/LedgerFormats.h>
+#include <ripple/protocol/SField.h>
+#include <ripple/protocol/TER.h>
+#include <ripple/protocol/TxFormats.h>
+#include <ripple/protocol/digest.h>
 #include <ripple/protocol/jss.h>
 #include <ripple/rpc/Context.h>
 #include <ripple/rpc/Role.h>
-#include <ripple/rpc/impl/TransactionSign.h>
+#include <boost/algorithm/string.hpp>
+#include <sstream>
 
 namespace ripple {
 
+class Definitions
+{
+private:
+    std::string
+    translate(std::string inp)
+    {
+        auto replace = [&](const char* f, const char* r) -> std::string {
+            std::string out = inp;
+            boost::replace_all(out, f, r);
+            return out;
+        };
+
+        auto find = [&](const char* s) -> bool {
+            return inp.find(s) != std::string::npos;
+        };
+
+        if (find("UINT"))
+        {
+            if (find("256") || find("160") || find("128"))
+                return replace("UINT", "Hash");
+            else
+                return replace("UINT", "UInt");
+        }
+
+        if (inp == "OBJECT")
+            return "STObject";
+        if (inp == "ARRAY")
+            return "STArray";
+        if (inp == "AMM")
+            return "AMM";
+        if (inp == "ACCOUNT")
+            return "AccountID";
+        if (inp == "LEDGERENTRY")
+            return "LedgerEntry";
+        if (inp == "NOTPRESENT")
+            return "NotPresent";
+        if (inp == "PATHSET")
+            return "PathSet";
+        if (inp == "VL")
+            return "Blob";
+        if (inp == "DIR_NODE")
+            return "DirectoryNode";
+        if (inp == "PAYCHAN")
+            return "PayChannel";
+        if (inp == "XCHAIN_BRIDGE")
+            return "XChainBridge";
+
+        std::string out;
+        size_t pos = 0;
+        for (;;)
+        {
+            pos = inp.find("_");
+            if (pos == std::string::npos)
+                pos = inp.size();
+            std::string token = inp.substr(0, pos);
+            if (token.size() > 1)
+            {
+                boost::algorithm::to_lower(token);
+                token.data()[0] -= ('a' - 'A');
+                out += token;
+            }
+            else
+                out += token;
+            if (pos == inp.size())
+                break;
+            inp = inp.substr(pos + 1);
+        }
+        return out;
+    };
+
+    Json::Value
+    generate()
```

Comment:
Why do you have this `generate` function? Why not place this functionality inside the `Definitions` constructor? (`generate` is not being invoked anywhere else)

Since `Definitions` need to be generated at most once in the life time of the program, is `Singleton` pattern appropriate for this requirement? Does it suffice to have one `static` instance of the `Definitions` object?

Reply:
There is only one `static` instance of the `Definitions` object - in the `doServerDefinitions` method.

Reply:
I understand -- but it might not be readily apparent to the readers of the code. I felt there are alternatives which are better poised to convey that there's only one instance

Reply:
I don't think that's necessary, since this is only a helper class and isn't used anywhere else.

Reply:
It's pretty easy to see the only place it's used is a static variable IMO - and it'd be a comparable search to finding the `getInstance` method. If you think it'd be easier I can move the `ServerDefinitions` implementation below the `doServerDefinitions` method, so that code is front-and-center?

Reply:
no, I think that is not required 👍